distributed/s3.py

```diff
@@ -486,58 +486,76 @@ def read_block_from_s3(filename, offset, length, s3pars={}, delimiter=None):
 
 
 @gen.coroutine
-def _read_text(fn, encoding='utf-8', errors='strict', lineterminator='\n',
-               executor=None, fs=None, lazy=True, collection=True):
-    from dask import do
+def _read_text(fn, keyname=None, encoding='utf-8', errors='strict', lineterminator='\n',
```

Comment:
Elsewhere we have `path` as the files location descriptor (part before first `/` is the bucket) - what is `keyname`'s function here?